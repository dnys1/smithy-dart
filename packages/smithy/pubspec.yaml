name: smithy
description: Smithy client runtime for Dart
version: 0.3.2
homepage: https://github.com/dnys1/smithy-dart
publish_to: https://pub.dillonnys.com

environment:
  sdk: ">=2.15.0 <3.0.0"

dependencies:
  aws_common:
    hosted: https://pub.dillonnys.com
    version: ^0.1.0
  built_collection: ^5.0.0
  built_value: ^8.0.0
  collection: ^1.15.0
  convert: ^3.0.1
  crypto: ^3.0.1
  fixnum: ^1.0.0
  http: ^0.13.0
  http_parser: ^4.0.0
  intl: ^0.17.0
  json_annotation: ">=4.4.0 <4.5.0"
  logging: ^1.0.2
  meta: ^1.7.0
  path: ^1.8.0
  retry: ^3.1.0
  shelf: ^1.1.0
  typed_data: ^1.3.0
  uuid: ^3.0.0
  xml: ^5.3.0

dev_dependencies:
  amplify_lints: ^1.0.0
  build_runner: ^2.0.0
  built_value_generator: ^8.1.3
  json_serializable: ^6.1.0
  test: ^1.16.0
