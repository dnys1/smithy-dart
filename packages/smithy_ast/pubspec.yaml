name: smithy_ast
description: Smithy AST model with serialization to and from JSON
version: 0.1.1
homepage: https://github.com/awslabs/smithy-dart
publish_to: https://unpub.dillonnys.com

environment:
  sdk: ">=2.15.0 <3.0.0"

dependencies:
  aws_common:
    hosted: https://unpub.dillonnys.com
    version: ^0.1.0
  built_collection: ^5.1.1
  built_value: ^8.1.3
  collection: ^1.15.0
  json_annotation: ^4.4.0
  meta: ^1.7.0

dev_dependencies:
  build_runner: ^2.1.7
  built_value_generator: ^8.1.3
  json_serializable: ^6.1.3
  lints: ^1.0.0
  test: ^1.16.0
