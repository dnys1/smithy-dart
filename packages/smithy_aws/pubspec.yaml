name: smithy_aws
description: Smithy runtime for AWS clients
version: 0.1.1
homepage: https://github.com/dnys1/smithy-dart
publish_to: https://pub.dillonnys.com

environment:
  sdk: ">=2.15.0 <3.0.0"

dependencies:
  aws_common:
    hosted: https://pub.dillonnys.com
    version: ^0.1.0
  aws_signature_v4:
    hosted: https://pub.dillonnys.com
    version: ^0.1.0
  built_collection: ^5.0.0
  built_value: ^8.0.0
  collection: ^1.15.0
  convert: ^3.0.1
  crclib: ^3.0.0
  crypto: ^3.0.1
  http: ^0.13.4
  intl: ^0.17.0
  json_annotation: ^4.4.0
  meta: ^1.7.0
  path: ^1.8.0
  smithy:
    hosted: https://pub.dillonnys.com
    version: ^0.1.0
  smithy_ast:
    hosted: https://pub.dillonnys.com
    version: ^0.1.0
  xml: ^5.3.1

# REMOVE FOR PUBLISHING
# dependency_overrides:
#   smithy:
#     path: ../smithy
#   smithy_ast:
#     path: ../smithy_ast

dev_dependencies:
  amplify_lints: ^1.0.0
  build_runner: ^2.1.7
  built_value_generator: ^8.0.0
  file: ^6.1.2
  glob: ^2.0.2
  json_serializable: ^6.1.4
  pubspec_parse: ^1.2.0
  smithy_codegen:
    path: ../smithy_codegen
  test: ^1.16.0
