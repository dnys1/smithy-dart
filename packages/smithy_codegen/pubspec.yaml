name: smithy_codegen
description: Smithy model generator from AST representation
version: 0.3.4
homepage: https://github.com/dnys1/smithy-dart
publish_to: https://pub.dillonnys.com

environment:
  sdk: ">=2.15.0 <3.0.0"

dependencies:
  args: ^2.3.0
  aws_common:
    hosted: https://pub.dillonnys.com
    version: ^0.1.0
  aws_signature_v4:
    hosted: https://pub.dillonnys.com
    version: ^0.1.0
  build_cli_annotations: ^2.0.0
  built_collection: ^5.1.1
  built_value: ^8.1.3
  code_builder: ^4.1.0
  collection: ^1.15.0
  dart_style: ^2.2.1
  fixnum: ^1.0.0
  grpc: ^3.0.2
  html2md: ^1.2.5
  http: ^0.13.4
  jmespath: ^2.0.0
  json_annotation: ">=4.4.0 <4.5.0"
  meta: ^1.7.0
  path: ^1.8.0
  protobuf: ^2.0.1
  pub_semver: ^2.1.0
  pubspec_parse: ^1.2.0
  shelf: ^1.1.0
  shelf_router: ^1.1.0
  smithy:
    hosted: https://pub.dillonnys.com
    version: ^0.3.0
  smithy_aws:
    hosted: https://pub.dillonnys.com
    version: ^0.3.0
  tuple: ^2.0.0
  uuid: ^3.0.0
  xml: ^5.3.0
  yaml_edit: ^2.0.1

# REMOVE FOR PUBLISHING
dependency_overrides:
  smithy:
    path: ../smithy
  smithy_aws:
    path: ../smithy_aws

dev_dependencies:
  amplify_lints: ^1.0.0
  build_cli: ^2.1.4
  build_runner: ^2.1.7
  built_value_generator: ^8.1.3
  build_verify: ^3.0.0
  build_version: ^2.1.0
  json_serializable: ^6.1.3
  smithy_test:
    path: ../smithy_test
  test: ^1.16.0

executables:
  smithy_codegen:
