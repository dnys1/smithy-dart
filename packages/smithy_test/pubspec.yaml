name: smithy_test
description: Test utilities for Smithy packages.
version: 1.0.0
publish_to: none

environment:
  sdk: ">=2.15.1 <3.0.0"

dependencies:
  http: ^0.13.4
  smithy_ast:
    path: ../smithy_ast
  smithy_codegen:
    path: ../smithy_codegen
  test: ^1.16.0

dev_dependencies:
  lints: ^1.0.0
