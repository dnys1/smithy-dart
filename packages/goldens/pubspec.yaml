name: goldens
description: Golden files for smithy-dart
version: 1.0.0
publish_to: none

environment:
  sdk: ">=2.14.4 <3.0.0"

dependencies:
  smithy:
    path: ../smithy

dependency_overrides:
  smithy_ast:
    path: ../smithy_ast
  smithy:
    path: ../smithy
  smithy_aws:
    path: ../smithy_aws
  smithy_codegen:
    path: ../smithy_codegen

dev_dependencies:
  file: ^6.1.2
  glob: ^2.0.2
  lints: ^1.0.0
  pubspec_parse: ^1.2.0
  smithy_codegen:
    path: ../smithy_codegen
  test: ^1.16.0
